src/Peer.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1); // @codeCoverageIgnore
+
+namespace Rinq;
+
+use Rinq\Exception\InvalidNamespaceException;
+use Rinq\Ident\PeerId;
+
+/*
+ * Peer represents a connection to a Rinq network.
+ *
+ * Peers can act as a server, responding to application-defined commands.
+ * Use Peer::listen() to start accepting incoming command requests.
+ *
+ * Command requests are sent by sessions, represented by the Session interface.
+ * Sessions can also send notifications to other sessions. Sessions are created
+ * by calling Peer::session().
+ *
+ * Each peer is assigned a unique ID, which is represented by the PeerID class.
+ * All IDs generated by the peer, such as session IDs and message IDs contain
+ * the peer ID, so that they can be traced to their origin easily.
+ */
+interface Peer
+{
+    /**
+     * The peer's unique identifier.
+     *
+     * @return PeerId The peer's unique identifier.
+     */
+    public function id(): PeerId;
+
+    /**
+     * Session creates and returns a new session owned by this peer.
+     *
+     * Creating a session does not perform any network IO. The only limit to the
+     * number of sessions is the memory required to store them.
+     *
+     * Sessions created after the peer has been stopped are unusable. Any
+     * operation will fail immediately.
+     *
+     * @return Session Newly created session.
+     */
+    public function session(): Session;
+
+    /**
+     * Listen starts listening for command requests in the given namespace.
+     *
+     * When a command request is received with a namespace equal to $namespace,
+     * the handler $handler is invoked.
+     *
+     * Repeated calls to listen() with the same namespace simply changes the
+     * handler associated with that namespace.
+     *
+     * $handler is invoked on its own routine for each command request.
+     *
+     * @param string         $namespace The namespace of the command request.
+     * @param CommandHandler $handler   The handler to fulfil the request.
+     *
+     * @return bool True If handler $handler starts listening in a new namespace.
```

Comment:
I think this boolean return information might be from one of the internal implementation detail classes, this method in the public API does not return anything.